src/api/middlewares/error-handlers.ts

```diff
@@ -0,0 +1,26 @@
+export interface CustomError {
+  name: string;
+  message: string;
+  status: number;
+}
+
+const createCustomError = (
+  message: string,
+  status: number,
+  name: string
+): CustomError => ({
+  name: name || "CustomError",
```

Comment:
if we think `name` might be empty, why not make it an optional parameter??

`const createCustomError = (
  message: string,
  status: number,
  name?: string
)`